trpcultivate_germplasm/src/Plugin/TripalImporter/GermplasmCrossImporter.php

```diff
@@ -0,0 +1,972 @@
+<?php
+
+namespace Drupal\trpcultivate_germplasm\Plugin\TripalImporter;
+
+use Drupal\Core\Entity\EntityTypeManager;
+use Drupal\Core\Messenger\MessengerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Render\Renderer;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\tripal\TripalImporter\Attribute\TripalImporter;
+use Drupal\tripal_chado\ChadoBuddy\PluginManagers\ChadoBuddyPluginManager;
+use Drupal\tripal_chado\Database\ChadoConnection;
+use Drupal\tripal_chado\Plugin\ChadoBuddy\ChadoCvtermBuddy;
+use Drupal\tripal_chado\Plugin\ChadoBuddy\ChadoPropertyBuddy;
+use Drupal\tripal_chado\TripalImporter\ChadoImporterBase;
+use Drupal\trpcultivate\Plugin\Validators\ValidDataFile;
+use Drupal\trpcultivate\Plugin\Validators\EmptyCell;
+use Drupal\trpcultivate\Plugin\Validators\GermplasmNameExists;
+use Drupal\trpcultivate\Plugin\Validators\ValidDelimitedFile;
+use Drupal\trpcultivate\Plugin\Validators\ValidHeaders;
+use Drupal\trpcultivate\Plugin\Validators\ValueInList;
+use Drupal\trpcultivate\TripalCultivateValidator\TripalCultivateValidatorManager;
+use Drupal\trpcultivate\Service\TripalCultivateFileTemplateService;
+use Drupal\trpcultivate\Service\ImportValidationHelper;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Tripal Cultivate Germplasm - Cross Importer.
+ *
+ * An importer for germplasm crosses developed in a breeding program.
+ *
+ * @TripalImporter(
+ *   id = "trpcultivate-germplasm-cross-importer",
+ *   label = @Translation("Tripal Cultivate: Germplasm Cross Importer"),
+ *   description = @Translation("Loads germplasm crosses into the system. This is useful for large datasets to ease the upload process."),
+ *   file_types = {"tsv"},
+ *   upload_description = @Translation("Please provide a data file."),
+ *   upload_title = @Translation("Germplasm Cross Data File*"),
+ *   use_analysis = FALSE,
+ *   require_analysis = FALSE,
+ *   use_button = True,
+ *   submit_disabled = FALSE,
+ *   button_text = "Import",
+ *   file_upload = TRUE,
+ *   file_local = FALSE,
+ *   file_remote = FALSE,
+ *   file_required = TRUE,
+ *   cardinality = 1,
+ *   menu_path = "",
+ *   callback = "",
+ *   callback_module = "",
+ *   callback_path = "",
+ * )
+ */
+#[TripalImporter(
+   id: 'trpcultivate-germplasm-cross-importer',
+   label: new TranslatableMarkup('Tripal Cultivate: Germplasm Cross Importer'),
+   description: new TranslatableMarkup('Loads germplasm crosses into the system. This is useful for large datasets to ease the upload process.'),
```

Comment:
Looking at the importer descriptions on the data loaders page, I'm thinking we should start expanding them :thinking: what do you think of something like this?

```suggestion
   description: new TranslatableMarkup('Creates germplasm cross pages associated with parental material through upload of a germplasm cross data file.'),
```

Reply:
Addressed in commit https://github.com/TripalCultivate/TripalCultivate-Germplasm/pull/33/commits/2a879eda09a0a6c5d16ce49a681412419305b5eb